Firebase/DynamicLinks/FDLURLComponents/FDLURLComponents.m

```diff
@@ -448,15 +449,37 @@ - (instancetype)init {
 
 @implementation FIRDynamicLinkComponents
 
+#pragma mark Deprecated Initializers.
 + (instancetype)componentsWithLink:(NSURL *)link domain:(NSString *)domain {
-  return [[self alloc] initWithLink:link domain:domain];
+  NSString *domainURIPrefix = [NSString stringWithFormat:@"https://%@", domain];
+  return [FIRDynamicLinkComponents componentsWithLink:link domainURIPrefix:domainURIPrefix];
 }
 
 - (instancetype)initWithLink:(NSURL *)link domain:(NSString *)domain {
+  NSString *domainURIPrefix = [NSString stringWithFormat:@"https://%@", domain];
+  return [self initWithLink:link domainURIPrefix:domainURIPrefix];
+}
+
+#pragma mark Initializers.
++ (instancetype)componentsWithLink:(NSURL *)link domainURIPrefix:(NSString *)domainURIPrefix {
+  if (![[domainURIPrefix lowercaseString] hasPrefix:@"https://"]) {
```

Comment:
Can we perform stricter checks up-front, not just prefix? There's a well-defined regex that determines what qualifies as a URL.

Reply:
Done.